text/0000-pose-graph-interface.md

```diff
@@ -0,0 +1,34 @@
+# Introduce a PoseGraphInterface Exported by MapBuilderInterface.
```

Comment:
"Introduce PoseGraphInterface to define public PoseGraph API." describes the change better.